cli/src/diff_util.rs

```diff
@@ -648,6 +663,67 @@ pub fn show_color_words_diff(
     Ok(())
 }
 
+pub fn show_file_by_file_diff(
```

Comment:
nit: is it possible (and make sense) to move this to `merge_tools/external.rs`?

(Maybe we can move `materialized_diff_stream()` to library, but I'm not sure about `diff_content()`.)

Reply:
I tried moving stuff around a bit, but the changes don't make it much cleaner.  Let me see if I can improve it on the next change in this sequence.